xwiki-platform-core/xwiki-platform-mentions/xwiki-platform-mentions-default/src/main/java/org/xwiki/mentions/internal/DefaultMentionsEventExecutor.java

```diff
@@ -189,13 +196,18 @@ public void consume()
                 DefaultMentionsEventExecutor.this.logger
                     .warn("Error during mention analysis of task [{}]. Cause [{}].", data, getRootCauseMessage(e));
                 if (data != null) {
-                    // push back the failed task at the beginning of the queue.
-                    try {
-                        DefaultMentionsEventExecutor.this.queue.put(data);
-                    } catch (InterruptedException interruptedException) {
+                    if (!data.isFailed()) {
+                        // push back the failed task at the beginning of the queue.
```

Comment:
The comment is a bit confusing because you're testing for not failed and the comment says it failed.